### ISSUE TITLE
Use `SourceSet` API to accurately collect source files

### ISSUE THREAD

**Issue by u-ways** [propactive/propactive#104]
## Background

It seems that #92 introduced an issue when loading the class required to generate the application property for package structured projects.

Upon initial analysis, it seems we have a test coverage gap for our class loader logic in which we only have unit tests, but no adequate / isolated / focused integration tests despite being a critical section of the plugin logic.

On top of that, we should rely on the [SourceSet API](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html) to get the correct directory for our property class loader.

> A `SourceSet` represents a logical group of Java source and resource files. They are covered in more detail in the [user manual](https://docs.gradle.org/current/userguide/building_java_projects.html#sec:java_source_sets).

This also applies to setting our default  `OutputDirectory`. We should rely on the main sources set configurations to find the correct `absolutePath`.

## Requirements

- [ ] Resolve the class loader bug for package structured projects.
- [ ] Update task output `destination` to default to main sources `resourcesDir`.
- [ ] Add integration tests that interact with packaged structured JVM projects.
- [ ] Using the `SourceSet` API will be a breaking change, release as a MAJOR version.
- [ ] Ensure task caching is not impacting.

In summary, rewrite how propactive tasks collect input and output files without impacting cache performance.

___
